src/custom/hooks/useUSDCPrice/index.ts

```diff
@@ -84,66 +83,78 @@ export default function useUSDCPrice(currency?: Currency) {
     return undefined
   }, [currency, stablecoin, v2USDCTrade, v3USDCTrade.trade]) */
 
-  useEffect(() => {
-    const supportedChain = supportedChainId(chainId)
-    if (!isQuoteLoading || !supportedChain || !sellTokenAddress || !sellTokenDecimals) return
+  const supportedChain = supportedChainId(chainId)
+  const baseAmount = supportedChain ? STABLECOIN_AMOUNT_OUT[supportedChain] : undefined
+  const stablecoin = baseAmount?.currency
+  const baseAmountRaw = baseAmount?.quotient.toString()
 
-    const baseAmount = STABLECOIN_AMOUNT_OUT[supportedChain]
-    const stablecoin = baseAmount.currency
+  const quoteParams = useMemo(() => {
+    if (!stablecoin || !supportedChain || !sellTokenAddress || !sellTokenDecimals) return null
 
-    const quoteParams = {
+    return {
       buyToken: stablecoin.address,
       sellToken: sellTokenAddress,
       kind: OrderKind.BUY,
-      amount: baseAmount.quotient.toString(),
+      amount: baseAmountRaw,
       chainId: supportedChain,
       fromDecimals: sellTokenDecimals,
       toDecimals: stablecoin.decimals,
       userAddress: account,
       // we dont care about validTo here, just use max
       validTo: MAX_VALID_TO_EPOCH,
     }
+  }, [account, baseAmountRaw, sellTokenAddress, sellTokenDecimals, stablecoin, supportedChain])
+
+  // SWR cache cow usd requests
+  const { data: priceResponse, error: errorResponse } = useSWR<CancelableResult<string | null> | null>(
```

Comment:
how often this cached result will be revalidated? when does it become stale?

Reply:
it's currently revalidated using default options, as shown:
<img width="844" alt="image" src="https://user-images.githubusercontent.com/21335563/169562146-f73d8e9e-9ac3-4e12-9eaf-546df235fc56.png">

i added custom options that are in a comment below